### PR TITLE
Expand workflow run with payload info

### PR DESCRIPTION
**Pull request by saraycp**
Extract some of the important information that is in the SCM webhook HTTP request (headers/body) into `WorkflowRun` attributes so they are easier to query.

### CODE REVIEW

**Comment by danidoni** [comment]
I'm not sure about the gains of extracting that extractor class...

**Comment by hennevogel** [comment]
> I'm not sure about the benefits of extracting that extractor class... 

What are you unsure about?

**Comment by danidoni** [comment]
> What are you unsure about?

We extract that class, but we don't use it anywhere else. I don't see any benefit in moving that code to it's own class.

**Comment by hennevogel** [comment]
>  I don't see any benefit in moving that code to it's own class.

Smaller class, easier re-use (see the data migration for instance).

**Comment by danidoni** [comment]
> > I don't see any benefit in moving that code to it's own class.
> 
> Smaller class, easier re-use (see the data migration for instance).
Which data migration? I'm a bit lost here... I only see it being used [here](https://github.com/openSUSE/open-build-service/pull/14527/files#diff-09911b866648b4d707f35444c891c4b39bf4a1804f4cf57b781f32fe1252ddddR2)

**Comment by hennevogel** [comment]
> Which data migration?

It's gone by now. So is the service.

**Comment by hennevogel** [comment]
Do not merge because it contains an "unsafe" data migration that probably needs to be run in the maintenance window

File: src/api/app/controllers/concerns/scm_webhook_headers_data_extractor.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module ScmWebhookHeadersDataExtractor
+  extend ActiveSupport::Concern
+
+  def set_scm_event
+    @gitlab_event = request.env['HTTP_X_GITLAB_EVENT']
+    # Gitea contains the Github headers as well, so we have to check that the Gitea ones are
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/controllers/concerns/scm_webhook_headers_data_extractor.rb:6]: But I'm just moving code around

**Comment by hennevogel** [comment]
Managed to move the migrations out to #14541

File: src/api/app/controllers/concerns/scm_webhook_headers_data_extractor.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/controllers/concerns/scm_webhook_headers_data_extractor.rb:6]: later maybe...

**Review by rubhanazeem — APPROVED** [review]

**Pull request merged**